### ISSUE TITLE
tree-view-sample file update watch() can't listen to the file properly

### ISSUE THREAD

**Issue by 1234567845246** [microsoft/vscode-extension-samples#865]
The FileSystemProvider class watch() function in the fileExplorer.ts file in the tree-view-sample cannot listen to the treeview properly, so the file tree cannot be updated after changes. In addition, the exists() file is deprecated and can be replaced with access(), and the fs module rm() and mkdir() can be used directly to create recursive creation files. mkdirp and rimraf packages have removed the interface from the example in high versions

**Comment by lucasnatanmelo** [comment]
Hello [1234567845246](https://github.com/1234567845246), did you successfully implement the create, edit, delete, rename and etc in folders and files using this FileSystemProvider Class or any other example in the tree view sample?

**Comment by 1234567845246** [comment]
I just use another 3rd party library "@parcel/watcher", this one works and treeview is changed. I didn't use the rest of the cases much. But this problem not only occurred to me, but also to my three classmates

**Comment by 1234567845246** [comment]
The FILE was successfully created and deleted from disk, but the treeview was not updated

**Comment by lucasnatanmelo** [comment]
Rigth, https://github.com/1234567845246. And did you succesfully implemented the functionality of drag and drop of files between folders, and the functionality of drag and drop the file for the code editing panel of VSCode? Can we contact or share your code to see how did you implmented these features? Thank you if you could help me!

**Comment by 1234567845246** [comment]
No, I'm working on how to add files and the input appears in the treeview, just like vscode itself Add files to the file tree.Sorry, I may not be able to help you
